ios/RNEngageBySailthru.m

```diff
@@ -0,0 +1,260 @@
+
+        if ([type isEqualToString:@"string"]) {
+            NSString *value = [[attributes valueForKey:key] valueForKey:@"value"];
+            [marAttributes setString:value forKey:key];
+
+        } else if ([type isEqualToString:@"stringArray"]) {
+            NSArray<NSString *> *value = [[attributes valueForKey:key] valueForKey:@"value"];
+            [marAttributes setStrings:value forKey:key];
+
+        } else if ([type isEqualToString:@"integer"]) {
+            NSNumber *value = [[attributes valueForKey:key] objectForKey:@"value"];
+            [marAttributes setInteger:[value integerValue] forKey:key];
+
+        } else if ([type isEqualToString:@"integerArray"]) {
+            NSArray<NSNumber *> *value = [[attributes valueForKey:key] valueForKey:@"value"];
+            [marAttributes setIntegers:value forKey:key];
+
+        } else if ([type isEqualToString:@"boolean"]) {
+            BOOL value = [[[attributes valueForKey:key] valueForKey:@"value"] boolValue];
+            [marAttributes setBool:value forKey:key];
+
+        } else if ([type isEqualToString:@"float"]) {
+            NSNumber *numberValue = [[attributes valueForKey:key] objectForKey:@"value"];
+            [marAttributes setFloat:[numberValue floatValue] forKey:key];
+
+        } else if ([type isEqualToString:@"floatArray"]) {
+            NSArray<NSNumber *> *value = [[attributes valueForKey:key] objectForKey:@"value"];
+            [marAttributes setFloats:value forKey:key];
+
+        } else if ([type isEqualToString:@"date"]) {
+            NSNumber *millisecondsValue = [[attributes valueForKey:key] objectForKey:@"value"];
+            NSNumber *value = @([millisecondsValue doubleValue] / 1000);
+
+            if (![value isKindOfClass:[NSNumber class]]) {
+                return;
+            }
+
+            NSDate *date = [NSDate dateWithTimeIntervalSince1970:[value doubleValue]];
+            if (date) {
+                [marAttributes setDate:date forKey:key];
+            } else {
+                return;
+            }
+
+        } else if ([type isEqualToString:@"dateArray"]) {
+            NSArray<NSNumber *> *value = [[attributes valueForKey:key] objectForKey:@"value"];
+            NSMutableArray<NSDate *> *dates = [[NSMutableArray alloc] init];
+            for (NSNumber *millisecondsValue in value) {
+                NSNumber *secondsValue = @([millisecondsValue doubleValue] / 1000);
+
+                if (![secondsValue isKindOfClass:[NSNumber class]]) {
+                    continue;
+                }
+
+                NSDate *date = [NSDate dateWithTimeIntervalSince1970:[secondsValue doubleValue]];
+                if (date) {
+                    [dates addObject:date];
+                }
+            }
+
+            [marAttributes setDates:dates forKey:key];
+        }
+    }
+
+    [[self engageBySailthruWithResolver:resolve rejecter:reject] setAttributes:marAttributes withResponse:^(NSError * _Nullable error) {
```

Comment:
This would be a good candidate for retrieving the `EngageBySailthru` instance at the start of the method and checking for nil - if it isn't supported we can return before doing all the attribute conversion.